contracts/CPKFactory.sol

```diff
@@ -31,30 +24,21 @@ contract CPKFactory {
     function createProxyAndExecTransaction(
         address owner,
         address safeVersion,
-        uint256 saltNonce,
+        uint256 saltNonceSalt,
         address fallbackHandler,
         bytes calldata execTxCalldata
     )
         external
         payable
         returns (bool execTransactionSuccess)
     {
-        bytes32 salt = keccak256(abi.encode(owner, saltNonce));
-
-        address payable proxy;
-        {
-            bytes memory deploymentData = abi.encodePacked(
-                type(GnosisSafeProxy).creationCode,
-                abi.encode(proxyImplSetter)
-            );
-            // solium-disable-next-line security/no-inline-assembly
-            assembly {
-                proxy := create2(0x0, add(0x20, deploymentData), mload(deploymentData), salt)
-            }
-            require(proxy != address(0), "create2 call failed");
-        }
+        bytes32 saltNonce = keccak256(abi.encode(owner, saltNonceSalt));
 
-        ProxyImplSetter(proxy).setImplementation(safeVersion);
+        gnosisSafeProxyFactory.createProxyWithNonce(
+            proxyImplSetter,
+            abi.encodeWithSelector(proxyImplSetter.setImplementation.selector, safeVersion),
```

Comment:
By doing this, the address of the proxy will still depend on the Safe implementation